lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -563,38 +564,24 @@ sub _generate_jobs {
 
             # variable expansion
             # replace %NAME% with $settings{NAME}
-            my $expanded;
-            do {
-                $expanded = 0;
-                for my $var (keys %settings) {
-                    my $val = $settings{$var};
-                    next unless ($val && $val =~ /(%\w+%)/);
-                    my $replace_var = $1;
-                    $replace_var =~ s/^%(\w+)%$/$1/;
-                    my $replace_val = $settings{$replace_var};
-                    next unless (defined $replace_val);
-                    $replace_val = '' if $replace_var eq $var;    #stop infinite recursion
-                    $val =~ s/%${replace_var}%/$replace_val/g;
-                    $settings{$var} = $val;
-                    $expanded = 1;
-                }
-            } while ($expanded);
+            my $obj          = OpenQA::Setting->new(%settings);
+            my $new_settings = $obj->replace_setting();
```

Comment:
After reading the code of `replace_settings` it looks like this would modify `$obj` itself rather than returning a new object. Therefore this assignment is irritating. I'd prefer: `my $new_settings = OpenQA::Setting->new(%settings)->replace_setting();`

Looking at this call I think that having an object for this is an overkill. (Since you already took the effort we can keep it but I actually prefer to keep things as simple as possible.)